### PR TITLE
Upgrade otel collector core, use the new Span Status code

### LINKED ISSUE #1841: Adjust Span status translations to match the specification
Once this is merged https://github.com/open-telemetry/opentelemetry-specification/pull/966 we need to adjust translation accordingly.

### PR DESCRIPTION
**Pull request by tigrannajaryan**
Resolves: https://github.com/open-telemetry/opentelemetry-collector/issues/1841

This uses the new Span Status Code/DeprecatedCode from core repo.

Changes to Span Status are described in more details in this core PR:
https://github.com/open-telemetry/opentelemetry-collector/pull/2031

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@kbrockhoff @anuraaga @axw @simitt @jalvz @AbhiPrasad @nilebox @james-bebbington @owais @jrcamp:
Codeowners of awsxrayexporter, awsxrayreceiver, sentryexporter, stackdriverexporter, sapmreceiver, elasticexporter please have a look. I am not confident that the changes I made correctly reflect the semantics of the corresponding vendor formats.

**Review by nilebox — APPROVED** [review]
LGTM for changes in Stackdriver exporter.

File: exporter/sentryexporter/sentry_exporter.go
Comment (AbhiPrasad) [inline exporter/sentryexporter/sentry_exporter.go:42]: Sentry relies on exact values for these status codes as per: https://github.com/getsentry/relay/blob/1a00926a6e8936173d4281777c27814d509a9fe5/relay-common/src/constants.rs#L254

To get around this, we should probably map `unset` and `error` to `unknown` for now. If/when Sentry accommodates for the otel spec change for span status, I can come back and update the exporter.

cc @rhcarvalho so Sentry folks are aware of this change.

**Review by axw — APPROVED** [review]
LGTM for elastic, thanks for checking!

**Review by AbhiPrasad — COMMENTED** [review]
Small thing about the Sentry Exporter, everything else LGTM.

File: exporter/sentryexporter/sentry_exporter.go
Comment (tigrannajaryan) [reply exporter/sentryexporter/sentry_exporter.go]: Done.

**Review by bogdandrutu — APPROVED** [review]

**Review by AbhiPrasad — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
I am going to merge this since it is blocking the release. Any fixes necessary in the affected exporters can be fixed in separate PRs that need to be submitted by the codeowners of the exporter.

**Pull request merged**
